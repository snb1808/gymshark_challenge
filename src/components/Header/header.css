.header-image {
    height: 3rem;
}

.mb-2 {
    margin-top: 0.2rem;
}

@media (max-width: 768px) {
    .mb-2 {
        margin-top: 1rem;
    }
}

.search-container {
    display: flex;
    flex-direction: row-reverse;
    width: -webkit-fill-available;
}

@media (max-width: 991px) {
    .search-container {
        width: 95vw;
        margin-top: 1rem;
    }
}

.form-inline {
    min-width: 20rem;
}

@media (max-width: 991px) {
    .form-inline {
        width: 100%;
    }
}

.mr-sm-2.form-control {
    width: 100%;
    margin: 0 !important;
}





