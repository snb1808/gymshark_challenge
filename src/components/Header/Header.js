import {
    Form,
    FormControl,
    Navbar,
    ToggleButton,
    ToggleButtonGroup,
} from 'react-bootstrap';
import logo from '../../Gymshark_logo.svg';
import './header.css';

const Header = ({ setGender, gender: selectedGender, filterExercises }) => {
    const genders = [{ gender: 'male', text: 'Men' }, { gender: 'female', text: 'Women' }];

    return (
        <Navbar bg="light" expand="lg" sticky='top'>
            <img aria-label='gymshark logo' className='header-image' src={logo}/>
            <Navbar.Brand>Exercise List</Navbar.Brand>
            <ToggleButtonGroup name='gender-toggle' className="mb-2">
                {genders.map(({ gender, text }) => (
                    <ToggleButton
                        type='radio'
                        variant='secondary'
                        name='radio'
                        value={gender}
                        checked={selectedGender === gender}
                        onChange={() => setGender(gender)}
                    >
                        {text}
                    </ToggleButton>
                ))}
            </ToggleButtonGroup>
            <div className='search-container'>
                <Form inline>
                    <FormControl
                        type="text"
                        placeholder="Search muscle to train"
                        className="mr-sm-2"
                        onChange={filterExercises}
                    />
                </Form>
            </div>
        </Navbar>
    );
}

export default Header;