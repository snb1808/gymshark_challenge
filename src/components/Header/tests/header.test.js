import React from 'react';
import Header from '../Header';
import { shallow } from 'enzyme';

describe('Header', () => {
   it('should render correctly', () => {
       const component = shallow(<Header />)

       expect(component.debug()).toMatchSnapshot();
   });
});