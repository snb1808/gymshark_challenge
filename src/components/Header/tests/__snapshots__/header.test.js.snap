// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
"<Navbar bg=\\"light\\" expand=\\"lg\\" sticky=\\"top\\" variant=\\"light\\" collapseOnSelect={false}>
  <img aria-label=\\"gymshark logo\\" className=\\"header-image\\" src=\\"Gymshark_logo.svg\\" />
  <NavbarBrand>
    Exercise List
  </NavbarBrand>
  <ForwardRef name=\\"gender-toggle\\" className=\\"mb-2\\" type=\\"radio\\" vertical={false}>
    <ToggleButton type=\\"radio\\" variant=\\"secondary\\" name=\\"radio\\" value=\\"male\\" checked={false} onChange={[Function: onChange]}>
      Men
    </ToggleButton>
    <ToggleButton type=\\"radio\\" variant=\\"secondary\\" name=\\"radio\\" value=\\"female\\" checked={false} onChange={[Function: onChange]}>
      Women
    </ToggleButton>
  </ForwardRef>
  <div className=\\"search-container\\">
    <Form inline={true}>
      <FormControl type=\\"text\\" placeholder=\\"Search muscle to train\\" className=\\"mr-sm-2\\" onChange={[undefined]} />
    </Form>
  </div>
</Navbar>"
`;
